File: SemestrWork/src/main/java/com/itis/servletsapp/dao/UsersRepository.java
package com.itis.servletsapp.dao;

import com.itis.servletsapp.dao.base.CrudRepository;
import com.itis.servletsapp.model.User;

import java.util.Optional;
//Not realized
public interface UsersRepository extends CrudRepository<User, Long> {
//    Optional<User> findByEmail(String email);
}

File: stalker-shop/src/main/java/com/itis/stalkershop/repositories/interfaces/UsersRepository.kt
package com.itis.stalkershop.repositories.interfaces;

import com.itis.stalkershop.models.User
import com.itis.stalkershop.models.UserDto
import java.util.*

interface UsersRepository: Repository<String, UserDto, User>  {
    fun findByEmail(email: String): Optional<User>
}
